PhysicsTools/TensorFlow/interface/Graph.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * TensorFlow graph interface.
+ * Based on TensorFlow C API 1.1.
+ *
+ * Author:
+ *   Marcel Rieger
+ */
+
+#ifndef PHYSICSTOOLS_TENSORFLOW_GRAPH_H
+#define PHYSICSTOOLS_TENSORFLOW_GRAPH_H
+
+#include <algorithm>
+#include <stdexcept>
+#include <string>
+
+#include <tensorflow/c/c_api.h>
+
+#include "PhysicsTools/TensorFlow/interface/Tensor.h"
+
+namespace tf
+{
+
+// forward declaration
+class Graph;
+
+// generic class containing all information of inputs to / ouptuts from a graph
+class GraphIO
+{
+public:
+    // constructur
+    GraphIO(Tensor* tensor, TF_Operation* tf_operation, const std::string& opName, int opIndex = 0);
+
+    // destructor
+    virtual ~GraphIO();
+
+private:
+    Tensor* tensor;
+    std::string opName;
+    int opIndex;
+
+    TF_Output tf_output;
+
+    friend Graph;
+};
+
+// the Graph class
+class Graph
+{
+public:
+    // constructor
+    Graph(const std::string& exportDir = "", const std::string& tag = "serve");
```

Comment:
I think it's safe to drop the `exportDir`. The default value of `tag` corresponds to a certain constant flag in TensorFlow (`tf.saved_model.tag_constants.SERVING`) which is default and recommended for graphs only used for serving purposes. Maybe it's helpful to keep the default to be consistent with the graph saving procedure in Python.